doctr/models/_utils.py

```diff
@@ -65,38 +65,29 @@ def extract_rcrops(
     """
     if boxes.shape[0] == 0:
         return []
-    if boxes.shape[1] != 5:
-        raise AssertionError("boxes are expected to be relative and in order (x, y, w, h, alpha)")
+    if boxes.shape[1] != 4:
+        raise AssertionError("boxes are expected to be relative and quadrangles")
 
     # Project relative coordinates
     _boxes = boxes.copy()
     height, width = img.shape[:2] if channels_last else img.shape[-2:]
     if _boxes.dtype != np.int:
-        _boxes[:, [0, 2]] *= width
-        _boxes[:, [1, 3]] *= height
+        _boxes[:, :, 0] *= width
+        _boxes[:, :, 1] *= height
 
     crops = []
-    # Determine rotation direction (clockwise/counterclockwise)
-    # Angle coverage: [-90°, +90°], half of the quadrant
-    clockwise = False
-    if np.sum(boxes[:, 2]) > np.sum(boxes[:, 3]):
-        clockwise = True
 
     for box in _boxes:
-        x, y, w, h, alpha = box.astype(dtype)
-        src_pts = cv2.boxPoints(((x, y), (w, h), alpha))[1:, :]
+        src_pts = box[1:, :]
         # Preserve size
-        if clockwise:
-            dst_pts = np.array([[0, 0], [w - 1, 0], [w - 1, h - 1]], dtype=dtype)
-        else:
-            dst_pts = np.array([[h - 1, 0], [h - 1, w - 1], [0, w - 1]], dtype=dtype)
+        _, (w, h), _ = cv2.minAreaRect(box)
```

Comment:
The thing is, we first need to extract crops to classify them and re-order points so that the crop is straight, so in this extraction function we have just 4 points and no information on the actual aspect ratio of the box